sklearn/neighbors/_nearest_centroid.py

```diff
@@ -205,30 +262,32 @@ def fit(self, X, y):
                     )
                 self.centroids_[cur_class] = X[center_mask].mean(axis=0)
 
-        if self.shrink_threshold:
+        # Compute within-class std_dev with unshrunked centroids
+        variance = np.square(X - self.centroids_[y_ind])
+        self.within_class_std_ = np.sqrt(variance.sum(axis=0) / (n_samples - n_classes))
+
+        if not is_X_sparse:
```

Comment:
I think I was able to make it work with both sparse and non-sparse matrices. Any way to improve the code block (below)?

```
if not is_X_sparse:
    if np.all(np.ptp(X, axis=0) == 0):
        raise ValueError("All features have zero variance. Division by zero.")
else:
    if np.all(np.ptp(X, axis=0).todense() == 0):
        raise ValueError("All features have zero variance. Division by zero.")
```

Reply:
In addition, any plan to add support for Array API to this module? I'm happy to make more future contributions so just let me know!

Reply:
Regarding the sparse support, I think the code is fine. I will review if we can make it better.

Regarding the Array API, we plan to add it to estimator that it is possible to do so. Here, we have some Cython code, so I don't think that it will be easy to use the array API but I did not yet check details regarding this part of improvement.